src/LHScriptX/Script.cpp

```diff
@@ -119,7 +120,12 @@ ScriptCommandParameter GetParameter(Token& argument)
 				const auto z = std::strtof(floatEnd + 1, &floatEnd);
 				if (static_cast<size_t>(floatEnd - str.c_str()) == static_cast<size_t>(str.length()))
 				{
-					const auto& island = Game::Instance()->GetLandIsland();
+					if (!Locator::terrainSystem::has_value())
+					{
+						throw std::runtime_error("Cannot get landscape before any are loaded");
```

Comment:
Repeating this conditional throw in lots of places smells. We don't check other locator patterns before usage. I think letting the actual error from accessing the value throw is better

Reply:
@bwrsandman Thoughts?

Reply:
This one is a valid error. We actually have a test to catch a `runtime_error`, not an assert: `LoadScene, load_abode_no_landscape`. This happens when you don't call `LOAD_LANDSCAPE` early enough. This is definitely something we want to check in this instance.

The others maybe not. I can remove them.